app/models/shipit/task.rb

```diff
@@ -177,27 +181,45 @@ def finished?
       !pending? && !running? && !aborting?
     end
 
-    def pid
-      pid = Shipit.redis.get("task:#{id}:pid")
-      pid.presence && pid.to_i
+    def ping
+      Shipit.redis.set(status_key, 'alive', ex: PRESENCE_CHECK_TIMEOUT)
     end
 
-    def pid=(pid)
-      Shipit.redis.set("task:#{id}:pid", pid, ex: 24.hour.to_i)
+    def alive?
+      Shipit.redis.get(status_key) == 'alive'
     end
 
-    def abort!(rollback_once_aborted: false)
-      target_pid = pid
-      return write("\nAbort: failed, PID unknown\n") unless target_pid.present?
+    def report_dead!
+      write("ERROR: Background job haven't reported back in #{PRESENCE_CHECK_TIMEOUT} seconds.")
+      error!
+    end
+
+    def should_abort?
```

Comment:
The idea if that it yield once every time you hit the abort button. Many command line utilities do a clean shutdown on first SIGTERM, and a qucik & dirty one on the second. I want every click of the abort button to be translated into a signal to the process.

Also passed the third click, we send SIGKILL instead of SIGTERM.